src/components/UserSelector.tsx

```diff
@@ -2,15 +1,47 @@
+import React, { useEffect, useState } from 'react';
+import { User } from '../types/User';
+import { getUsers } from './api/users';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+
+type Props = {
+  onSelectedUser: (val: User | null) => void;
+  selectedUser: User | null;
+  onSelectedPost: (val: Post | null) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  selectedUser,
+  onSelectedUser,
+  onSelectedPost,
+}) => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+
+  useEffect(() => {
+    getUsers().then(setUsers);
+  }, []);
+
+  const handleUserSelect = (user: User) => {
+    onSelectedPost(null);
+    onSelectedUser(user);
+  };
```

Comment:
You are calling `onSelectedPost(null)` and `onSelectedUser(user)` sequentially. It triggers two renders. It's better to combine these two updates into one using `useReducer` or `useState` with an object in the state.